lib/adapters/MochaAdapter.js

```diff
@@ -16,48 +20,51 @@ export default class MochaAdapter extends EventEmitter {
       runner.on('suite end', this.onSuiteEnd.bind(this))
       runner.on('end', this.onEnd.bind(this))
     })
-    this.idDataMap = {}
   }
 
   convertSuite (mochaSuite) {
     var name = mochaSuite.title
     var childSuites = []
     var tests = []
 
+    mochaSuite.index = this.index
+    this.index++
```

Comment:
I strongly prefer `this.index += 1`, makes the change for explicit and discourages inline increments (change var and use it at the same time)